.github/workflows/master.yml

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
+
+name: Master workflow
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build-dbtestapp:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build
+        run: |
+          cd mongoapi/dbtestapp/
+          go build ./...
+
+  docker-build-dbtestapp:
+    env:
+      DOCKER_REGISTRY: "docker.io/"
+      DOCKER_REPOSITORY: "omecproject/"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build and push Docker image
+        run: |
+          cd mongoapi/dbtestapp
+          make docker-build
+          make docker-push
```

Comment:
I think we may need to run `docker-push` only for `push` and not for `pull_request`, correct?
```suggestion
```

Reply:
I added a separate workflow for docker image and removed the line from master.yml.